### PR TITLE
Show `DataFrame` records in logs

### PR DESCRIPTION
**Pull request by neilbest-db**
See doc string in `DataFrameSyntax.scala`:

https://github.com/databrickslabs/overwatch/blob/e5dd9882ce0705b835aea3cfd147b09d690b3830/src/main/scala/com/databricks/labs/overwatch/utils/DataFrameSyntax.scala#L13-L99

### CODE REVIEW

File: src/main/scala/com/databricks/labs/overwatch/pipeline/SilverTransforms.scala [hunk 1]
```diff
@@ -1313,6 +1313,15 @@ trait SilverTransforms extends SparkSessionWrapper {
     // eagerly force this highly reused DF into cache()
     jobRunsLag30D.count()
 
+    
+    // TODO: remove or comment out or change log level or . . .
+
+    logger.log( Level.INFO, "Showing first 5 rows of `jobRunsLag30D`:")
```
Comment (sriram251-code) [inline src/main/scala/com/databricks/labs/overwatch/pipeline/SilverTransforms.scala:1319]: Why we r printing  5 rows of the data in logs?

File: src/main/scala/com/databricks/labs/overwatch/pipeline/SilverTransforms.scala [hunk 2]
```diff
@@ -1313,6 +1313,15 @@ trait SilverTransforms extends SparkSessionWrapper {
...
+
+    jobRunsLag30D
+      .showLines(5, 20, true)
+      .foreach( logger.log( Level.INFO, _))
+
```
Comment (sriram251-code) [inline src/main/scala/com/databricks/labs/overwatch/pipeline/SilverTransforms.scala:1324]: why we r printing it?

**Review by sriram251-code — CHANGES_REQUESTED** [review]

File: src/main/scala/com/databricks/labs/overwatch/utils/DataFrameSyntax.scala [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    def showLines( numRows: Int): Iterator[String] =
+      showLines( numRows, truncate = true)
+
+    def showLines( truncate: Boolean): Iterator[String] =
+      showLines( 20, truncate)
+
+    def showLines( numRows: Int, truncate: Boolean): Iterator[String] =
+      if( truncate) {
+        showLines( numRows, truncate = 20)
+      } else {
```
Comment (souravbaner-da) [inline src/main/scala/com/databricks/labs/overwatch/utils/DataFrameSyntax.scala:51]: @neilbest-db  why 20 count here?

File: src/main/scala/com/databricks/labs/overwatch/pipeline/SilverTransforms.scala (hunk 1 above)
Comment (neilbest-db) [reply src/main/scala/com/databricks/labs/overwatch/pipeline/SilverTransforms.scala:1319]: It's just to demonstrate a temporary debug output.  Any developer can use `.showLines()` anywhere & anyhow they want to during development and then remove it, comment it out, or conditionally report something at a higher log level — whatever.

File: src/main/scala/com/databricks/labs/overwatch/pipeline/SilverTransforms.scala (hunk 2 above)
Comment (neilbest-db) [reply src/main/scala/com/databricks/labs/overwatch/pipeline/SilverTransforms.scala:1324]: It provides an alternative to copy-pasting so much code into notebooks by providing feedback in the logs.  Any transformation is available at that point, naturally.

File: src/main/scala/com/databricks/labs/overwatch/utils/DataFrameSyntax.scala (hunk 3 above)
Comment (neilbest-db) [reply src/main/scala/com/databricks/labs/overwatch/utils/DataFrameSyntax.scala:51]: Like the `truncate` argument for [native](https://spark.apache.org/docs/3.3.0/api/scala/org/apache/spark/sql/Dataset.html#show(numRows:Int,truncate:Int,vertical:Boolean):Unit) `.show()` method it limits the column width of the formatted output.

Also note [this](https://github.com/databrickslabs/overwatch/issues/1227#issuecomment-2135408925):
> I have overloaded `.showLines()` with the same set of method signatures as Spark's `Dataset.show()` for API consistency.

**Comment by neilbest-db** [comment]
Pasted docstring into initial comment above for reference.

**Comment by neilbest-db** [comment]
@sriram251-code, do you need more info before we move ahead with this?  I don't think it's right for me to mark the conversations resolved . . . are you satisfied or still have doubts?

**Comment by sriram251-code** [comment]
I am fine with the comments

**Review by sriram251-code — APPROVED** [review]

**Comment by neilbest-db** [comment]
closes #1227

**Review by gueniai — CHANGES_REQUESTED** [review]
This needs to be controlled by a config (an Overwatch scoped spark config will do) to turn this logging on/off, with a default being false

**Comment by neilbest-db** [comment]
> This needs to be controlled by a config (an Overwatch scoped spark config will do) to turn this logging on/off, with a default being false

Done.  Please review.

**Review by gueniai — APPROVED** [review]

**Pull request merged**